apps/dashboard/src/main/java/com/akto/action/testing/MarketplaceAction.java

```diff
@@ -38,18 +41,65 @@ public String fetchTestingSources() {
     TestCategory category;
     Severity severity;
     String description;
+    List<String> tags;
+    String searchText;
+    List<TestSourceConfig> searchResults = new ArrayList<>();
+    private List<TestSubCategory> searchAktoTests;
+    private TestCategory[] categories;
+
     public String addCustomTest() {
         TestSourceConfig alreadyExists = TestSourceConfigsDao.instance.findOne("_id", url);
         if (alreadyExists != null) {
             addActionError("This test file has already been added");
             return ERROR.toUpperCase();            
         }
 
-        TestSourceConfig elem = new TestSourceConfig(url, category, subcategory, severity, description, getSUser().getLogin(), Context.now());
+        TestSourceConfig elem = new TestSourceConfig(url, category, subcategory, severity, description, getSUser().getLogin(), Context.now(),tags);
         TestSourceConfigsDao.instance.insertOne(elem);
         return Action.SUCCESS.toUpperCase();
     }
 
+    private void searchUtilityFunction(){
+        this.searchAktoTests = new ArrayList<>();
+        this.categories = GlobalEnums.TestCategory.values();
+        if(this.searchText == null){
+            for(TestSubCategory tsc : GlobalEnums.TestSubCategory.getValuesArray()){
```

Comment:
can we add some check in ui, if search text is null, keep showing existing response?

Reply:
did this, but can't change this in backend as for initial test filling we need this check. but for later blank searchText, the existing response is shown.